install-wrappers.sh

```diff
@@ -58,6 +64,45 @@ else
     esac
 fi
 
+if [ -n "${HOST_CLANG-}" ]; then
+    HOST_CLANG_EXE=$(command -v $HOST_CLANG)
+    HOST_CLANG_VER=$(echo "__clang_major__ __clang_minor__ __clang_patchlevel__" | $HOST_CLANG_EXE -E -P -x c - | xargs printf '%d.%d.%d')
+
+    mkdir -p $PREFIX/bin
+
+	# ex. /usr/lib/llvm-17/lib/clang/17
+    resdir=$($HOST_CLANG -print-resource-dir)
+	# ex. /usr/lib/llvm-17
+    llvmdir=${resdir%/lib/clang/*}
+	# ex /lib/clang/17
+    clangres=${resdir#$llvmdir}
+
+    mkdir -p $PREFIX$clangres
+
+    # link the header directory, prevent modification
+    ln -snf $resdir/include $PREFIX$clangres/include
+
+    # Note: clang will detect the "InstalledDir" based on the path that was used to invoke the tools
```

Comment:
Can you elaborate on this comment? In principle, won't this odd/unexpected feature essentially mean that the whole wrapper script that you generate below wouldn't be necessary? It would be good to make the comment clearer, to indicate that while this behaviour exists, we choose not to rely on it, because of reasons X and Y (which reasons?), and instead use a wrapper script. I guess the "This might still have some hidden effects" comments is that, somewhat, i.e. we could do that but choose not to since we don't know what implications it might have?

Just spelling it out a bit more explicitly would be helpful at least.

Reply:
The note is primary to myself, precisely because I dont know and cant explain if this might be an issue.

``` bash
%  clang -v                
Debian clang version 17.0.2 (1~exp1)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
Found candidate GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/12
Selected GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/12
Candidate multilib: .;@m64
Selected multilib: .;@m64
```

vs.

``` bash
%  /usr/lib/llvm-17/bin/clang -v
Debian clang version 17.0.2 (1~exp1)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/lib/llvm-17/bin
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/12
Selected GCC installation: /usr/lib/gcc/x86_64-linux-gnu/12
Candidate multilib: .;@m64
Selected multilib: .;@m64
```

Reply:
Left this open - anything to do here?
Its the way it is because I dont know better